projects/bird-atlas/src/app/+grid/grid-info/grid-info.component.ts

```diff
@@ -1,9 +1,156 @@
-import { Component } from '@angular/core';
+import { AfterViewInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, NgZone, OnDestroy, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { TableColumn } from '@swimlane/ngx-datatable';
+import LajiMap, { TileLayerName } from 'laji-map';
+import { datatableClasses } from 'projects/bird-atlas/src/styles/datatable-classes';
+import { BehaviorSubject, of, Subject } from 'rxjs';
+import { catchError, filter, map, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { AtlasApiService } from '../../core/atlas-api.service';
+import { BreadcrumbId, BreadcrumbService } from '../../core/breadcrumb.service';
+import { convertYkjToGeoJsonFeature } from '../../../../../laji/src/app/shared/service/coordinate.service';
+import { TranslateService } from '@ngx-translate/core';
+import { HeaderService } from 'projects/laji/src/app/shared/service/header.service';
+
+function getGeoJSONFeature(ykj: string) {
+  const langLngStr = ykj.split(':');
+  return convertYkjToGeoJsonFeature(langLngStr[0], langLngStr[1]);
+}
 
 @Component({
   selector: 'ba-grid-info',
-  template: ``
+  templateUrl: './grid-info.component.html',
+  styleUrls: ['./grid-info.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class GridInfoComponent {
+export class GridInfoComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('lajiMap', { static: false }) lajiMapElem: ElementRef;
+
+  private unsubscribe$ = new Subject<void>();
+  private loadMap$ = new BehaviorSubject<string>(undefined);
+
+  data$ = this.route.paramMap.pipe(
+    tap(() => {
+      this.breadcrumbs.setBreadcrumbName(BreadcrumbId.GridInfo, undefined);
+      this.loading = true;
+      this.cdr.markForCheck();
+    }),
+    switchMap(params => this.atlasApi.getGridElement(params.get('id'))),
+    catchError(() => of(undefined)),
```

Comment:
I think it's little confusing to have `catchError` in the middle, how about this kind of structure
```
...,
switchMap(params => this.atlasApi.getGridElement(params.get('id')).pipe(
    map(<map function>),
    tap(<tap function)
),
catchError(() => ({
      elem: undefined,
      rows: [],
      status: 404
    })
),
tap(() => {
     // this is needed in both cases
    this.loading = false;
    this.cdr.markForCheck();
}));
```

Reply:
👍